src/source/src/parser/json_parser.rs

```diff
@@ -105,6 +105,10 @@ mod tests {
         for payload in [
             br#"{"i32":1,"bool":true,"i16":1,"i64":12345678,"f32":1.23,"f64":1.2345,"varchar":"varchar","date":"2021-01-01","timestamp":"2021-01-01 16:06:12.269","decimal":12345.67890}"#.as_slice(),
             br#"{"i32":1,"f32":12345e+10,"f64":12345,"decimal":12345}"#.as_slice(),
+            br#"{"amount": "1111111111111111111160"}"#.as_slice(),
```

Comment:
The field names need to correspond to a field in "descs" passed to the  `builder: SourceStreamChunkBuilder`

Reply:
Else it will be `None`

Reply:
Seems like it is still parsed first even if it does not correspond to any field, rather than entirely ignored.

Reply:
Logs:
```
2023-01-17T08:31:23.995279Z  WARN risingwave_source::connector_source: message parsing failed Protocol error: InvalidNumber at character 202 ('6'), skipping
```

Corresponding position in json
```
"amount": 6467408489094391385660
                             ^
```

With the source:
```
CREATE SOURCE eth_erc20_event_source (
"sysTimestamp" bigint
) with (...) ROW FORMAT JSON;
```

Reply:
yeah, that is unfortunately unavoidable atm

ingesting as string and then casting is indeed the only workaround.